src/Features/CSharp/Portable/InitializeParameter/CSharpAddParameterCheckCodeRefactoringProvider.cs

```diff
@@ -15,31 +16,33 @@ namespace Microsoft.CodeAnalysis.CSharp.InitializeParameter
     internal class CSharpAddParameterCheckCodeRefactoringProvider :
         AbstractAddParameterCheckCodeRefactoringProvider<
             ParameterSyntax,
-            BaseMethodDeclarationSyntax,
             StatementSyntax,
             ExpressionSyntax,
             BinaryExpressionSyntax>
     {
+        protected override bool IsFunctionDeclaration(SyntaxNode node)
+            => InitializeParameterHelpers.IsFunctionDeclaration(node);
+
         protected override SyntaxNode GetTypeBlock(SyntaxNode node)
             => node;
 
-        protected override SyntaxNode GetBody(BaseMethodDeclarationSyntax containingMember)
-            => InitializeParameterHelpers.GetBody(containingMember);
+        protected override IBlockOperation GetBlockOperation(SyntaxNode functionDeclaration, SemanticModel semanticModel, CancellationToken cancellationToken)
+            => InitializeParameterHelpers.GetBlockOperation(functionDeclaration, semanticModel, cancellationToken);
 
-        protected override void InsertStatement(SyntaxEditor editor, BaseMethodDeclarationSyntax methodDeclarationSyntax, SyntaxNode statementToAddAfterOpt, StatementSyntax statement)
-            => InitializeParameterHelpers.InsertStatement(editor, methodDeclarationSyntax, statementToAddAfterOpt, statement);
+        protected override void InsertStatement(SyntaxEditor editor, SyntaxNode functionDeclaration, IMethodSymbol method, SyntaxNode statementToAddAfterOpt, StatementSyntax statement)
+            => InitializeParameterHelpers.InsertStatement(editor, functionDeclaration, method, statementToAddAfterOpt, statement);
 
         protected override bool IsImplicitConversion(Compilation compilation, ITypeSymbol source, ITypeSymbol destination)
             => InitializeParameterHelpers.IsImplicitConversion(compilation, source, destination);
 
         protected override bool CanOffer(SyntaxNode body)
         {
-            if (body is ArrowExpressionClauseSyntax arrowExpressionClauseSyntax)
+            if (InitializeParameterHelpers.IsExpressionBody(body))
             {
-                return arrowExpressionClauseSyntax.TryConvertToStatement(
-                    semicolonToken: SyntaxFactory.Token(SyntaxKind.SemicolonToken), 
-                    createReturnStatementForExpression: false, 
-                    statement: out var _);              
+                return InitializeParameterHelpers.TryConvertExpressionBodyToStatement(body,
+                    semicolonToken: SyntaxFactory.Token(SyntaxKind.SemicolonToken),
+                    createReturnStatementForExpression: false,
+                    statement: out var _);
```

Comment:
do we have a test that exercises this codepath when TryConvertExpressionBodyToStatement returns false?

Reply:
yes: https://github.com/Neme12/roslyn/blob/957882b0ce32707c6540872dbebb51c4cf5eb03e/src/EditorFeatures/CSharpTest/InitializeParameter/AddParameterCheckTests.cs#L1170

Reply:
great.  thanks!